### DISCUSSION TITLE
Recalculating Start Time and Event Duration

### DISCUSSION THREAD

**Discussion by juzz197** [cpvalente/ontime#646] [Q&A]
I'm involved in running live events and theatre.  Our live events, by their nature, are highly dynamic which is why I've been hunting for something like ontime - and I'm blown away by its functionality already!

One thing I can't seem to solve at the moment is dynamically updating event start times - but not sure if I'm missing something!

There are two scenarios we usually have:
1) Changes to the running order before and during the show.  I'd like to be able to move an event and see the event that's moved and the subsequent events update their start times dynamically.
2) Many events are not fixed times - they can under and overrun (hosting, audience interaction).   I would manually move to the next event but this won't cause the subsequent start times to update.

It's really useful for our performers to see how the event is running to be prepared and also to ensure we don't breach any venue curfews!

I could see this implemented as a tickbox by the start time in the event that 'Start time is last end time' - similar to the option when an event is created - but would force a update of the start time if the event is moved or after any previous event finishes.  Alternatively a manual update button could be used.

It'd also be great to see the user defined fields exposed in the editor, as we would use them for potentially dynamic information - but happy to use a second window/tab on the cue sheet view for that.

Thanks for the great work!

**Comment by cpvalente** [comment]
Hi @juzz197 , these are very interesting questions, we have considered some variations of these cases and have some ongoing work. It would be interesting to hear your thoughts:

> 1. Changes to the running order before and during the show. I'd like to be able to move an event and see the event that's moved and the subsequent events update their start times dynamically.

We have work in progress for linking events, this means that an event start or end time could be linked to the previous or next event.
So:

1 - Warm up: `09:00` - `09:30`
2 - Pre-drinks: `link-previous` - `10:00`
3 - Show start: `10:00` - `11:00`

If you were to drag a new event between 1 and 2, the start of pre-drink would become the new end time.
Nice! But there are some caveats to consider
- When reordering or creating events, you ned to be conscious of potentially affecting another event
- What would happen if the event in between has accidentally a finish time that is after the finish time of pre-drinks (eg: 2.5 - Last minute: `link-previous` - `11:00`

Most users suggestions so far has been that linked events become locked, so you wouldnt be able to add or move in between, thus not serving your use case any longer.

> Many events are not fixed times - they can under and overrun (hosting, audience interaction). I would manually move to the next event but this won't cause the subsequent start times to update.

I havent personally considered this but I think there is a concept missing. The times of events in runtime is what we consider "Scheduled Time". Unless you are running in Roll Mode they dont actually mean anything for the app.
We have work in progress to show the user an Actual Start and Actual End, so that you can keep track of the differences between the schedule and the runtime.

However this doesnt seem to completely fit your use case
What I understand you need is some more fine grained distinctions on time, ie:
- an event needs to start at 10:30 but can finish whenever
- an event needs to finish at 10:30 but it starts whenever the one before finishes
- an event needs to last 5 minutes but we dont know when it starts and it finishes

The short answer is: none of this is currently possible in the app. But if we can produce a clear specification we increase the chances of it being implemented in a closer future

Look forwards to hearing your thoughts

**Comment by juzz197** [comment]
Thanks for the feedback!

Sounds like some of the enhancements may partially address some of my requirements and I look forward to testing them - I have a non-production system where I can do this.

Linked events will certainly help.  In general, for my events, start time derives from the end time of the last event and end time is start time + duration.  This is where the scheduled vs. actual might be useful.  But I also need to consider that duration of an event will fall in to two general categories:
1) fixed duration - known performance length e.g. song
2) Variable - could be due to an event restart (e.g. technical problem)/non-fixed duration where I add an estimated time to the event (as in the hosting/comedy set/audience interaction mentioned before) and manually advance when the event actually finishes.

I think it would be good for me to try these features once they are implemented, I'd be happy to help define specifications that address any gaps that hopefully will be useful to those working on live performance/theatre.

Excited to work with you on this!

**Reply by cpvalente** [reply]
Hi @juzz197 , thank you for this.
We are still planning the initial scope for v3, but I will make sure to take your feedback in consideration.

We will make a beta release of ontime v3 as soon as we have a stable direction

**Reply by juzz197** [reply]
Great news - if I can help with confirming/defining any requirements, please let me know.
